src/modules/ProductLicense.rb

```diff
@@ -1144,12 +1144,24 @@ def AskLicenseAgreement(src_id, dir, patterns, action, enable_back, base_product
       end
 
       licenses_ref = arg_ref(licenses)
-      DisplayLicenseDialog(
+
+      label = Pkg::SourceGeneralData(src_id)["name"]
+      title = _("License Agreement")
+
+      if !label.empty?
```

Comment:
Alternatively you can use `%{}` expansion:

```ruby
repository_name = Pkg::SourceGeneralData(src_id)["name"]
...
_("%{repository_name} License Agreement") % {repository_name: repository_name}
```

In that case the `TRANSLATORS` comment is not needed as the expansion is self descriptive.